rest-server/test/userManager.js

```diff
@@ -0,0 +1,388 @@
+              'value': '8507b5d862306d5bdad95b3d611b905ecdd047b0165ca3905db0d861e76bce8f3a8046e64379e81f54865f7c41b47e57cec887e5912062211bc9010afea8ab12',
+              'modifiedIndex': 7,
+              'createdIndex': 7
+            }],
+          'modifiedIndex': 8,
+          'createdIndex': 8
+        }
+      });
+
+
+    nock(etcdHosts)
+      .put('/v2/keys/users/new_user/passwd?prevExist=true', { 'value': '055ce1d842cf891d62d2bbedfbdcd3e1a9421ac94f4109fc01e11bd99b029671358bd0a871c7857d2ac01aa528c2504a05bcfc190205cc164d8f4c05cc5808e4' })
+      .reply(200, {
+        'action': 'update',
+        'node': {
+          'key': '/users/new_user/passwd',
+          'value': '055ce1d842cf891d62d2bbedfbdcd3e1a9421ac94f4109fc01e11bd99b029671358bd0a871c7857d2ac01aa528c2504a05bcfc190205cc164d8f4c05cc5808e4',
+          'modifiedIndex': 9,
+          'createdIndex': 9
+        },
+        'prevNode': {
+          'key': '/users/new_user/passwd',
+          'value': '8507b5d862306d5bdad95b3d611b905ecdd047b0165ca3905db0d861e76bce8f3a8046e64379e81f54865f7c41b47e57cec887e5912062211bc9010afea8ab12',
+          'modifiedIndex': 10,
+          'createdIndex': 10
+        }
+      });
+
+      nock(etcdHosts)
+      .put('/v2/keys/users/new_user/admin', { 'value': 'false' })
+      .reply(200, {
+        'action':'update',
+        'node':{
+          'key':'/users/new_user/admin',
+          'value':'false',
+          'modifiedIndex':11,
+          'createdIndex':11
+        },
+        'prevNode':{
+          'key':'/users/new_user/admin',
+          'value':'true',
+          'modifiedIndex':12,
+          'createdIndex':12
+        }
+      });
+
+  });
+
+  //
+  // Positive cases
+  //
+
+  it('Case 1 (Positive): update user password.', (done) => {
+    global.chai.request(global.server)
+      .put('/api/v1/user')
+      .set('Authorization', 'Bearer ' + validToken)
+      .send(JSON.parse(global.mustache.render(newUserTemplate, { 'username': 'new_user', 'password': 'abcdef', 'admin': true, 'modify': true })))
+      .end((err, res) => {
+        global.chai.expect(res, 'status code').to.have.status(201);
+        global.chai.expect(res, 'response format').be.json;
+        global.chai.expect(res.body.message, 'response message').equal('update successfully');
+        done();
+      });
+  });
+
+  //
+  // Positive cases
+  //
+
+  it('Case 2 (Positive): update user set admin=false.', (done) => {
+    global.chai.request(global.server)
+      .put('/api/v1/user')
+      .set('Authorization', 'Bearer ' + validToken)
+      .send(JSON.parse(global.mustache.render(newUserTemplate, { 'username': 'new_user', 'password': 'abcdef', 'admin': false, 'modify': true })))
+      .end((err, res) => {
+        global.chai.expect(res, 'status code').to.have.status(201);
+        global.chai.expect(res, 'response format').be.json;
+        global.chai.expect(res.body.message, 'response message').equal('update successfully');
+        done();
+      });
+  });
+
```

Comment:
Add negative cases? Update a non-existing user, etc. And what if password is set to an empty string?

Reply:
if password sets to an empty string will trigger parameter validation error. "password\" is not allowed to be empty.
Already add two negative case:1) update a non-existing user 2)update password to be empty